### PR TITLE
Fix obfuscation failure and merge Proguard rules

### PR DESCRIPTION
**Pull request by Goooler**
## Description

Release build failed due to missing rules about `com.skydoves.balloon.**`.

## Type of change

:white_check_mark: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: proguard.pro [hunk 1]
```diff
@@ -1,54 +1,24 @@
...
+-dontpreverify
+-dontusemixedcaseclassnames
+-dontskipnonpubliclibraryclasses
+-verbose
+-keepattributes InnerClasses, *Annotation*, Signature, SourceFile, LineNumberTable
+
+-keep class com.android.** { *; }
+
+-keepclasseswithmembernames class * {
+    native <methods>;
```
Comment (Goooler) [inline proguard.pro]: Move these rules from `proguard-android-optimize.txt`.

File: proguard.pro [hunk 2]
```diff
@@ -1,54 +1,24 @@
--keep,allowshrinking,allowoptimization class com.android.launcher3.** {
-  *;
```
Comment (Goooler) [inline proguard.pro:2]: These rules are located in `proguard.flags`, we can just load them.

File: proguard.pro [hunk 3]
```diff
@@ -81,85 +53,29 @@
...
-}
-
--keep class com.android.** {
-  *;
-}
+-keep class com.google.protobuf.Timestamp { *; }
 
-# Keep Smartspacer's client SDK
+# TODO: Remove this after the change in https://github.com/KieronQuinn/Smartspacer/pull/58 has been released.
```
Comment (MrSluffy) [inline proguard.pro:47]: Let's wait wen it released and remove this, since we're not yet gonna release soon

**Comment by Goooler** [comment]
We can verify release builds for PRs and commits, just drop the debug builds.

**Review by MrSluffy — APPROVED** [review]

**Pull request merged**